designs/2019-allow-all-directives-in-line-comments/README.md

```diff
@@ -0,0 +1,112 @@
+
+## Drawbacks
+
+I don't see any drawbacks to this change.
+
+## Backwards Compatibility Analysis
+
+### Latent Directives
+
+Users may have added directives to their code using line comments and then never noticed that they had no effect. With this change, they will now begin functioning again. In some cases this may be welcome, since the user's original intention would finally be respected. In other cases the code may have changed since the user added the latent directive. In that case this change may cause an unwanted directive to be enabled.
+
+Anecdotally, I did an audit of a large swath of our code base and discovered only two directives which were being ignored due to their comment type. In both cases, enabling would be the correct behavior.
+
+### Accidental Directives
+
+Users may also have written inline comments which were not intended as a directive, but parse as directives. I found one example in ESLint's own codebase:
+
+> `lineNumTokenBefore = 0; // global return at beginning of script`
+
+-- [newline-before-return.js:173](https://github.com/eslint/eslint/blob/02d7542cfd0c2e95c2222b1e9e38228f4c19df19/lib/rules/newline-before-return.js#L137)
+
+This would result in the following, somewhat confusing, error after upgrading to a version that contained this change:
+
+```
+/Users/jeldredge/projects/eslint/lib/rules/newline-before-return.js
+  137:51  error  'return' is defined but never used     no-unused-vars
+  137:58  error  'at' is defined but never used         no-unused-vars
+  137:61  error  'beginning' is defined but never used  no-unused-vars
+  137:71  error  'of' is defined but never used         no-unused-vars
+  137:74  error  'script' is defined but never used     no-unused-vars
+
+✖ 5 problems (5 errors, 0 warnings)
+```
+
+## Alternatives
+
+An earlier revision of this RFC suggested reporting an error/warning when a directive that is only supported in a block comment was found in a line comment. See the following section for a discussion of why this may be a worth doing in addition to the changes detailed above.
+
+## Optional Short Term Improvements
+
+Since code bases may contain line comments which were not intended as directives, but would be parsed as such after this change, (see [Accidental Directives](#accidental-directives) above) this change will be a breaking change and thus will need to wait for a major release. If the next major release is far off we could employ a short term fix of making line comment directives warnings which could be shipped in a minor release.
```

Comment:
Another possibility would be to make the changes to all comments but `// global` for now, and leave global to the next major release.

Reply:
I don't think `// global` is the only example. If someone writes `// eslint is stupid`, then that will attempt to be parsed as a configuration of the rule `is` (and will fail due to lack of colon).

Reply:
Fair enough. I'll remove this comment.